### PR TITLE
GH-323 - Add extra task fields to advanced and full text search

### PR DESCRIPTION
**Pull request by evdhoudt**
Fixes NCI-Agency/anet#323

### CODE REVIEW

File: client/src/components/AdvancedSearch.js [hunk 1]
```diff
@@ -17,6 +17,32 @@ import {Person, Task, Position} from 'models'
...
+		Organization: <OrganizationFilter
+			queryKey="responsibleOrgId"
+			queryIncludeChildOrgsKey="includeChildrenOrgs"
+		/>,
+		Status: <SelectSearchFilter
+			queryKey="status"
+			values={["ACTIVE", "INACTIVE"]}
+			labels={["Active", "Inactive"]}
+		/>,
+		'Project status': <SelectSearchFilter
```
Comment (VassilIordanov) [inline client/src/components/AdvancedSearch.js]: Should be hidden if customEnum is null

File: src/main/resources/migrations.xml [hunk 2]
```diff
@@ -869,10 +869,10 @@
 			<column name="plannedCompletion" type="datetime" />
 		</addColumn>
 		<addColumn tableName="tasks">
-			<column name="customField" type="text" />
+			<column name="customField" type="nvarchar(max)" />
```
Comment (gjvoosten) [inline src/main/resources/migrations.xml]: Has this migration already run on staging/production? Then you should provide a new migration that alters this field. You can't really modify historic migrations.

File: src/main/resources/migrations.xml [hunk 3]
```diff
@@ -869,10 +869,10 @@
...
 		</addColumn>
 		<addColumn tableName="tasks" >
-			<column name="customFieldEnum" type="text" />
+			<column name="customFieldEnum" type="nvarchar(255)" />
```
Comment (gjvoosten) [inline src/main/resources/migrations.xml]: Likewise.

File: src/main/java/mil/dds/anet/search/mssql/MssqlTaskSearcher.java [hunk 4]
```diff
@@ -34,12 +34,13 @@ public TaskList runSearch(TaskSearchQuery query, Handle dbHandle) {
 		
 		String text = query.getText();
 		if (text != null && text.trim().length() > 0) { 
-			whereClauses.add("(CONTAINS((longName), :text) OR shortName LIKE :likeQuery)");
+			whereClauses.add("(CONTAINS((longName), :text) OR CONTAINS((customField), :text) OR shortName LIKE :likeQuery)");
```
Comment (gjvoosten) [inline src/main/java/mil/dds/anet/search/mssql/MssqlTaskSearcher.java]: Use a single `CONTAINS((longName, customField), :text)`.

File: src/main/java/mil/dds/anet/search/mssql/MssqlTaskSearcher.java [hunk 5]
```diff
@@ -53,22 +54,52 @@ public TaskList runSearch(TaskSearchQuery query, Handle dbHandle) {
...
+			args.put("projectedCompletionStart", Utils.handleRelativeDate(query.getProjectedCompletionStart()));
+		}
+
+		if (query.getProjectedCompletionEnd() != null) {
+			whereClauses.add("projectedCompletion <= :projectedCompletionEnd");
+			args.put("projectedCompletionEnd", Utils.handleRelativeDate(query.getProjectedCompletionEnd()));
+		}
+
+		if (query.getCustomField() != null) {
+			whereClauses.add("[customField] LIKE :customField");
```
Comment (gjvoosten) [inline src/main/java/mil/dds/anet/search/mssql/MssqlTaskSearcher.java]: Why the square brackets (unneeded here)?

File: src/main/java/mil/dds/anet/search/mssql/MssqlTaskSearcher.java [hunk 6]
```diff
@@ -53,22 +54,52 @@ public TaskList runSearch(TaskSearchQuery query, Handle dbHandle) {
...
+			args.put("customField", query.getCustomField());
```
Comment (gjvoosten) [inline src/main/java/mil/dds/anet/search/mssql/MssqlTaskSearcher.java]:
```
args.put("customField", Utils.prepForLikeQuery(query.getCustomField()) + "%");
```

**Pull request merged**